db/system_keyspace.cc

```diff
@@ -1357,12 +1357,31 @@ schema_ptr system_keyspace::legacy::aggregates() {
     return schema;
 }
 
-future<> system_keyspace::setup_version(sharded<netw::messaging_service>& ms) {
+future<system_keyspace::local_info> system_keyspace::load_local_info() {
+    auto msg = co_await execute_cql(format("SELECT host_id, cluster_name FROM system.{} WHERE key=?", LOCAL), sstring(LOCAL));
+
+    local_info ret;
+    if (!msg->empty()) {
+        auto& row = msg->one();
+        if (row.has("host_id")) {
+            ret.host_id = locator::host_id(row.get_as<utils::UUID>("host_id"));
```

Comment:
untyped_result_set::get_opt, and make the members optional so we can tell if we were able to load or not?

Reply:
In the absence of c++23's `std::optional<T>::transform()` this is going to be explicit if (or ?:) anyway

Reply:
@tchaikov will probably succeed in convincing me to allow C++23 before ratification.